### DISCUSSION TITLE
Setting Configurations in Container

### DISCUSSION THREAD

**Discussion by andrewdunndev** [tensorchord/pgvecto.rs#345] [General]
Was looking to make changes to runtime [resource consumption parameters](https://www.postgresql.org/docs/current/runtime-config-resource.html#RUNTIME-CONFIG-RESOURCE) for a deployment of pgvecto.rs on a beefy machine. However I've been looking at the postgres upstream documentation.

Are these parameters relevant for pgvecto.rs?

Can these be set at instance time with the container, via environmental variable or by adding on to the command like `-c shared_buffers=8G`?

**Comment by VoVAllen** [comment]
What's your machine memory and how many photos do you have? pgvecto.rs manages the memory separate from pg. Therefore too big shared_buffers may cause memory contention between pg and pgvecto.rs.

**Reply by andrewdunndev** [reply]
I was talking with @mertalev about my instance of [immich](https://github.com/immich-app/immich) which is ~2T and ~200k assets. Mainly I was looking for things I could fiddle with that are outside of the immich codebase which is working to standardize around your stable release and likely will be changing a lot of the performance things that I'm hitting now.

I've got plenty of memory, I currently limit the pod the immich containers are in to 64G but I don't see utilization anywhere near that.

**Reply by VoVAllen** [reply]
200k is pretty easy to fit in the memory. The memory needed by pgvecto.rs should be less than 4G.
